src/Client.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\NotificationClient;
+
+use Closure;
+use GuzzleHttp\Client as GuzzleClient;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\GuzzleException;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\MessageFormatter;
+use GuzzleHttp\Middleware;
+use GuzzleHttp\Psr7\Request;
+use JsonException;
+use Keboola\NotificationClient\Exception\ClientException as NotificationClientException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Range;
+use Symfony\Component\Validator\Constraints\Url;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Symfony\Component\Validator\Validation;
+
+abstract class Client
+{
+    private const DEFAULT_USER_AGENT = 'Notification PHP Client';
+    private const DEFAULT_BACKOFF_RETRIES = 3;
+    private const JSON_DEPTH = 512;
+
+    /** @var GuzzleClient */
+    protected $guzzle;
+    /** @var LoggerInterface */
+    // phpcs:ignore
+    private $logger;
```

Comment:
At hledam, jak hledam, ta logger property se nikde nepouziva
```suggestion
```

Reply:
No proto tam mám ten ignore :)

Reply:
No asi se to na nic nepoužije